File: erlang/darts/src/darts.erl
-module(darts).

-export([score/2]).

score(X, Y) ->
    Distance = distance_from_origin(X, Y),
    do_score(Distance).

do_score(Distance) when Distance > 10 -> 0;
do_score(Distance) when Distance > 5 -> 1;
do_score(Distance) when Distance > 1 -> 5;
do_score(_Distance) -> 10.

distance_from_origin(X, Y) -> math:sqrt(X * X + Y * Y).

File: haskell/darts/src/Darts.hs
module Darts
  ( score,
  )
where

score :: Float -> Float -> Int
score x y
  | distance > 10 = 0
  | distance > 5 = 1
  | distance > 1 = 5
  | otherwise = 10
  where
    distance = sqrt (x * x + y * y)
